deployment/rolling.json

```diff
@@ -0,0 +1,110 @@
+{
+    "kind": "DeploymentConfig",
+    "apiVersion": "v1",
+    "metadata": {
+        "name": "hooks",
+        "creationTimestamp": null,
+        "labels": {
+            "name": "mysql"
+        }
+    },
+    "spec": {
+        "strategy": {
+            "type": "Rolling",
+            "recreateParams": {
+                "pre": {
+                    "failurePolicy": "Retry",
+                    "execNewPod": {
+                        "command": [
+                            "/bin/true"
+                        ],
+                        "env": [
+                            {
+                                "name": "VAR",
+                                "value": "pre-deployment"
+                            }
+                        ],
+                        "containerName": "mysql-55-centos7"
+                    }
+                },
+                "post": {
+                    "failurePolicy": "Ignore",
+                    "execNewPod": {
+                        "command": [
+                            "/bin/false"
+                        ],
+                        "env": [
+                            {
+                                "name": "VAR",
+                                "value": "post-deployment"
+                            }
+                        ],
+                        "containerName": "mysql-55-centos7"
+                    }
+                }
+            },
+            "rollingParams": {
+                "updatePeriodSeconds": 1,
+                "intervalSeconds": 1,
+                "timeoutSeconds": 600
+            },
+            "resources": {}
+        },
+        "triggers": [
+            {
+                "type": "ConfigChange"
+            }
+        ],
+        "replicas": 1,
+        "selector": {
+            "name": "mysql"
+        },
+        "template": {
+            "metadata": {
+                "creationTimestamp": null,
+                "labels": {
+                    "name": "mysql"
+                }
+            },
+            "spec": {
+                "containers": [
+                    {
+                        "name": "mysql-55-centos7",
+                        "image": "openshift/mysql-55-centos7:latest",
```

Comment:
centos image is in docker.io and it is more convenient to use it without manually pull the image. BTW, this is for testing deployment rolling update function, I think it is ok to use centos image.